modules/alloy/src/main/ts/ephox/alloy/alien/Descend.ts

```diff
@@ -1,13 +1,15 @@
-import { Struct } from '@ephox/katamari';
+import { Fun } from '@ephox/katamari';
 import { Node, Text, Traverse, Element } from '@ephox/sugar';
 
 export interface ElementAndOffset {
-  element: () => Element;
-  offset: () => number;
+  readonly element: () => Element;
+  readonly offset: () => number;
 }
 
-const point: (element: Element, offset: number) => ElementAndOffset =
-  Struct.immutable('element', 'offset');
+const point = (element: Element, offset: number): ElementAndOffset => ({
+  element: Fun.constant(element),
```

Comment:
This isn't a public API, so should we unthunk all the places `Struct.immutable` was used as well? That should help with size more, but maybe that's a task for another day 🤷‍♀

Reply:
Done. This one didn't have much impact.

Reply:
Knocked a few bytes off.

Reply:
I actually meant all places sorry, not just `ElementAndOffset`. Though like the other one, that's probably a much larger task. Thanks either way 👍